tests/test_transforms.py

```diff
@@ -988,608 +997,367 @@ def _test_projection(lhs: Optional[UnboundPredicate[L]], rhs: Optional[UnboundPr
         raise ValueError(f"Comparing unrelated: {lhs} <> {rhs}")
 
 
+def _assert_projection_strict(
+    pred: BooleanExpression,
+    transform: Transform[S, T],
+    expected_type: type[BooleanExpression],
+    expected_human_str: Optional[str] = None,
+) -> None:
+    result = transform.strict_project(name="name", pred=pred)
+
+    assert type(result) is expected_type or AlwaysFalse
```

Comment:
nit: I found it a bit confusing that we were defining the expected_type as `AlwaysFalse` when actually `None` was being returned by `strict_project`, but maybe the alternative of making `expected_type` `Optional` is more confusing 🤔

Reply:
I agree, let me fix that in a follow up PR 👍